### PR TITLE
(Cart) Show sample details

### PR DESCRIPTION
**Pull request by joshsadam**
## Description of changes

When the label for a sample is clicked, instead of going to the sample page, the details for the sample are displayed in a "Drawer" on the right hand side of the page.  Currently it displays the basic sample details and all metadata associated with it.  Future plans could include editing the metadata, displaying sequence files, etc ...

sample-details

## Related issue

#192 

## Checklist
The following will be completed when final merge is complete

~~Things for the developer to confirm they've done before the PR should be accepted:~~

* [ ] ~~CHANGELOG.md (and UPGRADING.md if necessary) updated with information for new change.~~
* [ ] ~~Tests added (or description of how to test) for any new features.~~
* [ ] ~~User documentation updated for UI or technical changes.~~

### CODE REVIEW

**Comment by wwhsiao** [comment]
This is really cool!

Cheers,

Will

William Hsiao, PhD
Senior Scientist, BCCDC Public Health Laboratory
Clinical Assistant Professor, Pathology & Laboratory Medicine, UBC
Adjunct Professor, Molecular Biology and Biochemistry, SFU
Rm 4066, 655 West 12th Avenue
Vancouver, BC, V5Z 4R4
Canada
Tel: 604-707-2561
Fax: 604-707-2603

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/samples/dto/SampleDetails.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
+package ca.corefacility.bioinformatics.irida.ria.web.samples.dto;
+
+import java.util.Map;
+
+import ca.corefacility.bioinformatics.irida.model.sample.MetadataTemplateField;
+import ca.corefacility.bioinformatics.irida.model.sample.Sample;
+import ca.corefacility.bioinformatics.irida.model.sample.metadata.MetadataEntry;
+
+/**
+ * Used to based details of a {@link Sample} back to the user interface.
```
[the remark this answers is not in the document]
Comment (joshsadam) [reply src/main/java/ca/corefacility/bioinformatics/irida/ria/web/samples/dto/SampleDetails.java]: Thanks, fixed in c97d7d75aaff4be942e8f3524c96a1ee4095645e

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/samples/SamplesController.java [hunk 2]
```diff
@@ -144,6 +146,20 @@ public String getSampleSpecificPage(final Model model, @PathVariable Long sample
...
+	 * Get {@link Sample} details for a specific sample.
+	 *
+	 * @param id {@link Long} identifier for a sample.
+	 * @return {@link SampleDetails} for the {@link Sample}
+	 */
+	@RequestMapping(value = "/samples", produces = MediaType.APPLICATION_JSON_VALUE)
+	@ResponseBody
+	public SampleDetails getSampleDetails(@RequestParam Long id) {
+		Sample sample = sampleService.read(id);
+		boolean modifiable = this.isProjectManagerForSample(sample);
```
[the remark this answers is not in the document]
Comment (joshsadam) [reply src/main/java/ca/corefacility/bioinformatics/irida/ria/web/samples/SamplesController.java]: Good idea, updated in d196003126eaf7105a4f5b9e76cfff1c45f01fc7

File: src/main/webapp/pages/cart.html [hunk 3]
```diff
@@ -15,48 +15,18 @@
             emptySub:
               /*[[#{cart.emptySub}]]*/ "Add some samples to it to make it happy :)",
             imageAlt: /*[[#{cart.imageAlt}]]*/ "Empty Cart"
+          },
+          sampleDetails: {
```
[the remark this answers is not in the document]
Comment (joshsadam) [reply src/main/webapp/pages/cart.html]: Added in f3a8c3a4c82ba444c5e685c56541f2aa537db703

**Comment by apetkau** [comment]
Thanks @joshsadam. Everything looks great.

**Pull request merged**